QUICKSTART.md

```diff
@@ -1,136 +1,211 @@
-gcp_credentials_json["private_key"] = bytes([_a ^ _b for _a, _b in zip(base64.b64decode(gcp_credentials_json["private_key"]), b"quickstart-sv"*150)]).decode("utf-8")
-credential = GCPPipelineCredentials.from_services_dict(gcp_credentials_json, schema_prefix)
+```
 
-# if you re-use an edited schema, then uncomment this part, so you can save it to file
-# schema_file_path = "examples/schemas/quickstart.yml"
+d. Name your table
+```
+parent_table = 'json_doc'
+```
 
+e. Specify your schema file location
+```
+schema_file_path = "schema.yml"
 ```
 
-### 2. Create a pipeline
+f. Load credentials
+```
+f = open('credentials.json')
+gcp_credentials_json = json.load(f)
+f.close()
+
+# Private key needs to be decoded (because we don't want to store it as plain text)
+gcp_credentials_json["private_key"] = bytes([_a ^ _b for _a, _b in zip(base64.b64decode(gcp_credentials_json["private_key"]), b"quickstart-sv"*150)]).decode("utf-8")
+credentials = GCPPipelineCredentials.from_services_dict(gcp_credentials_json, schema_prefix)
 ```
-pipeline = Pipeline(schema_name)
-pipeline.create_pipeline(credential)
 
-# Optionally uncomment to re-use a schema
-# schema = Pipeline.load_schema_from_file(schema_file_path)
-# pipeline.create_pipeline(credential, schema=schema)
+## 5. Create a DLT pipeline
 
+a. Instantiate a pipeline
+```
+pipeline = Pipeline(schema_name)
 ```
 
-### 3. Pass the data to the pipeline and give it a table name.
+b. Create the pipeline with your credentials
+```
+pipeline.create_pipeline(credentials)
 ```
 
-rows = [{"name":"Ana", "age":30, "id":456, "children":[{"name": "Bill", "id": 625},
-                                                       {"name": "Elli", "id": 591}
-                                                      ]},
+## 6. Load the data from the JSON document
 
-        {"name":"Bob", "age":30, "id":455, "children":[{"name": "Bill", "id": 625},
-                                                       {"name": "Dave", "id": 621}
-                                                      ]}
-       ]
+a. Load JSON document into a dictionary
+```
+f = open('data.json')
+data = json.load(f)
+f.close()
+```
 
-pipeline.extract(iter(rows), table_name=parent_table)
+## 7. Pass the data to the DLT pipeline
 
-# Optionally the pipeline to un-nest the json into a relational structure
-pipeline.unpack()
+a. Extract the dictionary into a table
+```
+pipeline.extract(iter(data), table_name=parent_table)
+```
 
-# Optionally save the schema for manual edits/future use.
-# schema = pipeline.get_default_schema()
-# schema_yaml = schema.as_yaml()
-# f = open(data_schema_file_path, "a")
-# f.write(schema_yaml)
-# f.close()
+b. Unpack the pipeline into a relational structure
+```
+pipeline.unpack()
+```
 
+c. Save schema to `schema.yml` file
 ```
+schema = pipeline.get_default_schema()
+schema_yaml = schema.as_yaml()
+f = open(schema_file_path, "a")
+f.write(schema_yaml)
+f.close()
+```
+
+*question: do we really want to append to the schema file?*
```

Comment:
@rudolfix should this be a write permission instead?

Reply:
this should be improved
1. in `as_yaml` set `remove_default_hints` to `True` to get clear schema
2. use context manager `with open... as f` to open the file
3. indeed open file with "w"